### PR TITLE
Checkout: Add new PayPal payment method that uses PayPal JS

### PR DESCRIPTION
**Pull request by sirbrillig**
## Background

The PayPal Express NVP payment method system is being deprecated and we need to transition to using the PayPal JS library and the PPCP order system for our PayPal transactions. See https://github.com/Automattic/payments-shilling/issues/3039

## Proposed Changes

In this PR we add a new PayPal payment method to replace the old one. The new method uses PayPal's JS library to display the submit button and take appropriate actions.

> [!NOTE]
> This only changes the PayPal button in checkout. It does not yet change the button on the "Add new payment method"/"Change payment method" form. This is to reduce the number of changes in one PR.

> [!IMPORTANT]
> This does not enable the new payment method by default except in development. It leaves the PayPal Express payment method as-is until a the `checkout/paypal-ppcp` config flag is turned on.

Requires D166389-code

## Testing Instructions

- Apply D166389-code and point `public-api.wordpress.com` at your sandbox. You will also need to set some configuration on your sandbox; see that diff's testing instructions for more info.
- Make sure your currency is in USD.
- Use `calypso.localhost` on this branch when testing; calypso.live links will not work because the new payment method is disabled there.
- Add a product to your cart and visit checkout.
- Click "Edit" (if required) on the payment method step of checkout and select PayPal.
- Click to submit the transaction.
- Verify that you remain on checkout but see a new PayPal window pop up over checkout (this differs from PayPal Express which does a full-page redirect).
- Confirm the transaction inside the popup window (you will need to use a sandboxed PayPal account). The window should close when you're done.
- Verify that checkout redirects to the post-checkout page(s).
- Verify that the transaction is successful with no errors.

Also we need to make sure that if D166389-code is _not_ applied, this PR has no effect and PayPal continues to work.

### CODE REVIEW

**Comment by martian77** [comment]
I think the diff at https://github.com/Automattic/payments-shilling/issues/3236 should have the necessary front end to receive the approval information, we 'just' need to update this to send the original order id and the PayPal Order ID when the 'order approved' response arrives.

File: client/my-sites/checkout/src/hooks/use-create-payment-methods/index.tsx [hunk 1]
```diff
@@ -49,21 +50,28 @@ const debug = debugFactory( 'calypso:use-create-payment-methods' );
...
+export function useCreatePayPalExpress( {
 	labelText,
 	shouldShowTaxFields,
 }: {
 	labelText?: string | null;
 	shouldShowTaxFields?: boolean;
-} ): PaymentMethod {
+} ): PaymentMethod | null {
 	const store = useMemo( () => createPayPalStore(), [] );
+	const shouldUsePayPalPPCP = isEnabled( 'checkout/paypal-ppcp' );
```
Comment (sirbrillig) [inline client/my-sites/checkout/src/hooks/use-create-payment-methods/index.tsx]: Note to self: PP Express and PPCP have different payment method IDs. If this feature flag is enabled and the endpoint does _not_ return the PPCP payment method as available but _does_ return PP Express, then we should probably show that instead.

Comment (sirbrillig) [reply client/my-sites/checkout/src/hooks/use-create-payment-methods/index.tsx]: Done.

**Review by JessBoctor — COMMENTED** [review]
Since code-D166389 has been deployed, all I had to do was add the `define( 'PAYPAL_PPCP_ENABLED', true);` constant to my Sandbox to test. 

When I have the constant set to true and I click on the PayPal submit, a _new tab_ is opened in the browser and I am prompted to sign into PayPal. 

I am not sure what I did different, but my first couple of tries resulted in a failure and a `PayPal transaction not approved` message.

I used my `testoctor` account and you can see the orders with the `redirected-to-paypal-ppcp` status in Payments Admin > Orders

However, after toggling the `PAYPAL_PPCP_ENABLED` on and off, the new version is working as described. Though I am still seeing a new tab open, not a in-window pop-up.

When `PAYPAL_PPCP_ENABLED` is disabled, the original browser window is redirected to PayPal. I am able to select my payment method from the wallet, then I am redirected back to Checkout and the transaction is approved.

Update: After re-reading the testing instructions, it seems a new window is the expected behavior. I must have misread that the first time. Looks like everything is working as expected!

**Review by JessBoctor — APPROVED** [review]

**Comment by sirbrillig** [comment]
Yup! Sorry for not being clear. The PayPal Express experience should be a full-page redirect, and the PayPal PPCP experience should be opening a new window above checkout that closes when the payment is confirmed.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17011645

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Hi @sirbrillig, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include this string: `The PayPal transaction was not approved.`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17011645).

**Pull request merged**